src/stores/referenceStore.ts

```diff
@@ -115,32 +123,60 @@ class ReferenceStore extends BaseStore {
     }
   };
 
+  findAppropriateDescriptionById = (
+    descriptions: any[],
+    assetId: string
+  ): DescriptionResponseModel => {
+    return (
+      descriptions.find(
+        (rawDescription: any) => rawDescription.Id === assetId
+      ) || {}
+    );
+  };
+
+  @action
   fetchAssets = () => {
-    return this.api
-      .fetchAll()
-      .then((resp: any) => {
-        if (resp) {
-          const dtoAssets = resp.Assets || resp;
-          if (!dtoAssets) {
-            return;
-          }
-          runInAction(() => {
-            this.assets = dtoAssets.map((x: any) =>
-              mappers.mapToAsset(x, this.categories)
+    const requests = [this.api.fetchAll(), this.api.fetchAssetsDescriptions()];
+
+    return Promise.all(requests).then(data => {
+      const assets = data[0].Assets || data[0];
+      this.descriptions = data[1].Descriptions || data[1];
+      if (assets && assets.length > 0) {
+        runInAction(() => {
+          this.assets = assets.map((rawAsset: AssetResponseModel) => {
+            const appropriateDescription = this.findAppropriateDescriptionById(
+              this.descriptions,
+              rawAsset.Id
+            );
+            return mappers.mapToAsset(
+              rawAsset,
+              this.categories,
+              appropriateDescription
             );
           });
-        }
-        return Promise.resolve();
-      })
-      .catch(Promise.reject);
+        });
+      }
+      return Promise.resolve();
+    });
   };
 
   fetchAssetById = (id: string) => {
-    return this.api.fetchAssetById(id).then((resp: any) => {
-      const dtoAsset = resp.Asset || resp;
-      const asset = mappers.mapToAsset(dtoAsset, this.categories);
-      this.assets.push(asset);
-      return Promise.resolve(asset);
+    return this.api.fetchAssetById(id).then(asset => {
+      let mappedAsset;
+      asset = asset.Asset || asset;
```

Comment:
Maybe it make sense to push result into new variable with different name just for better readability?